test/exml_query_tests.erl

```diff
@@ -61,6 +61,21 @@ no_element_with_ns_query_test() ->
                  exml_query:subelement_with_ns(chat_marker(),
                                                <<"wrong">>, none)).
 
+element_with_attr_query_test() ->
+    ?assertEqual(xml(<<"<received xmlns='urn:xmpp:chat-markers:0'
+                                id='0047ee62-9418-4ef8-abd8-0d08e4140b72'/>)">>),
+                 exml_query:subelement_with_attr(chat_marker(), <<"xmlns">>,
+                                                 <<"urn:xmpp:chat-markers:0">>)),
+
+    ?assertEqual(xml(<<"<received xmlns='urn:xmpp:chat-markers:0'
```

Comment:
We should also test the case when there are more than one sub-element in the message.
The test only test the happy-way scenario.
I.e. if we modify element_with_attr to just return first child, the tests would not spot the change.